### PR TITLE
chore: update examples in docs to async/await

### LINKED ISSUE #525: Convert examples in docs to async/await
The examples in the docs from this repo need to be updated to use async/await.

### PR DESCRIPTION
**Pull request by PedroMiguelSS**
This PR converts `interface-ipfs-core` docs examples to use async/await syntax.

Closes #525.

### CODE REVIEW

File: SPEC/OBJECT.md [hunk 1]
```diff
@@ -66,14 +66,10 @@ const obj = {
...
 
-ipfs.object.put(obj, (err, cid) => {
-  if (err) {
-    throw err
-  }
-  console.log(cid.toString())
-  // Logs:
-  // QmPb5f92FxKPYdT3QNBd1GKiL4tZUXUrzF4Hkpdr3Gf1gK
-})
+const cid = await ipfs.object.pu(obj)
```
Comment (alanshaw) [inline SPEC/OBJECT.md]:
```suggestion
const cid = await ipfs.object.put(obj)
```

File: SPEC/OBJECT.md [hunk 2]
```diff
@@ -164,23 +156,21 @@ A great source of [examples][] can be found in the tests for this API.
 
 - `enc`, the encoding of multihash (base58, base64, etc), if any.
 
-`callback` must follow `function (err, links) {}` signature, where `err` is an error if the operation was not successful and `links` is an Array of [DAGLink](https://github.com/vijayee/js-ipfs-merkle-dag/blob/master/src/dag-node.js#L199-L203) objects.
+**Returns**
 
-If no `callback` is passed, a [promise][] is returned.
+| Type | Description |
+| -------- | -------- |
+| `Promise<Array>` | An Array of [DAGLink](https://github.com/vijayee/js-ipfs-merkle-dag/blob/master/src/dag-node.js#L199-L203) objects |
```
Comment (alanshaw) [inline SPEC/OBJECT.md]: Please update to https://github.com/ipld/js-ipld-dag-pb/blob/master/src/dag-link/dagLink.js

File: SPEC/OBJECT.md [hunk 3]
```diff
@@ -164,23 +156,21 @@ A great source of [examples][] can be found in the tests for this API.
...
-    throw err
-  }
-  console.log(links)
-  // Logs:
-  // []
-})
+const links = await ipfs.object.links(multihash)
+console.log(links)
+// Logs:
+// []
```
Comment (alanshaw) [inline SPEC/OBJECT.md]: Can you please update this example to include actual content!

File: SPEC/PIN.md [hunk 4]
```diff
@@ -8,55 +8,62 @@
...
 
 ```JavaScript
-ipfs.pin.ls(function (err, pinset) {
-  if (err) {
-    throw err
-  }
-  console.log(pinset)
-})
+const pinset = await ipfs.pin.ls()
+console.log(pinset)
```
Comment (alanshaw) [inline SPEC/PIN.md]: Can we have some commented example content here please?

File: SPEC/PIN.md [hunk 5]
```diff
@@ -8,55 +8,62 @@
...
 ```
 
-If no `callback` is passed, a promise is returned.
-
 **Example:**
 
 ```JavaScript
-ipfs.pin.add(hash, function (err) {})
+const pinset = await ipfs.pin.add(hash)
+console.log(pinset)
```
Comment (alanshaw) [inline SPEC/PIN.md]: Can we have some commented example content here please?

File: SPEC/PUBSUB.md [hunk 6]
```diff
@@ -10,30 +10,29 @@
 
 > Subscribe to a pubsub topic.
 
-##### `ipfs.pubsub.subscribe(topic, handler, [options], [callback])`
+##### `ipfs.pubsub.subscribe(topic, handler, [options])`
 
 - `topic: String`
 - `handler: (msg) => {}` - Event handler which will be called with a message object everytime one is received. The `msg` has the format `{from: String, seqno: Buffer, data: Buffer, topicIDs: Array<String>}`.
 - `options: Object` - (Optional) Object containing the following properties:
   - `discover: Boolean` - (Default: `false`) Will use the DHT to find other peers.
```
Comment (alanshaw) [inline SPEC/PUBSUB.md]: Please could you please add a comment that this option is currently unimplemented?

**Comment by alanshaw** [comment]
ping @PedroMiguelSS

File: SPEC/BLOCK.md [hunk 7]
```diff
@@ -65,44 +58,40 @@ if no options are passed, it defaults to `{ format: 'dag-pb', mhtype: 'sha2-256'
...
-  // Logs:
-  // a serialized object
-  console.log(block.cid.toBaseEncodedString())
-  // Logs:
-  // the CID of the object
-})
+console.log(block.data.toString())
+// Logs:
+// a serialized object
+console.log(block.cid.toBaseEncodedString())
```
[the remark this answers is not in the document]
Comment (PedroMiguelSS) [reply SPEC/BLOCK.md]: Sure thing! I'll update `block` and `dag` examples.

File: SPEC/OBJECT.md (hunk 1 above)
Comment (PedroMiguelSS) [reply SPEC/OBJECT.md]: Nice catch!! Thanks.

File: SPEC/PUBSUB.md (hunk 6 above)
Comment (PedroMiguelSS) [reply SPEC/PUBSUB.md]: If `discover` was the only option and it's not implemented yet, it means that currently there are no options available. Right?

File: SPEC/OBJECT.md (hunk 3 above)
Comment (PedroMiguelSS) [reply SPEC/OBJECT.md]: Use a `cid` that actually has links so that we can demonstrate using the `console.log()`. Am I right?

**Comment by PedroMiguelSS** [comment]
Do you prefer one PR per file/API?

File: SPEC/PUBSUB.md (hunk 6 above)
Comment (alanshaw) [reply SPEC/PUBSUB.md]: Correct, but please leave this here as we need to document `onError` asap, which you can pass to the HTTP client for dealing with pubsub errors and disconnections.

Any chance you could send a PR for it?

**Comment by alanshaw** [comment]
Thanks @PedroMiguelSS ❤️

**Pull request merged**